tutorials/submitit.ipynb

```diff
@@ -0,0 +1,7550 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Hyperparameter Optimization on Slurm via SubmitIt\n",
+    "\n",
+    "This notebook serves as a quickstart guide for using the Ax library with the SubmitIt library in an ask-tell loop. [SubmitIt](https://github.com/facebookincubator/submitit/) is a Python toolbox for submitting jobs to [Slurm](https://slurm.schedmd.com/quickstart.html). \n",
+    "\n",
+    "The notebook demonstrates how to use the Ax client in an ask-tell loop where each trial is scheduled to run on a Slurm cluster asynchronously.\n",
+    "\n",
+    "To use this script, run it on a slurm node either as an interactive notebook or export it as a Python script and run it as a Slurm job.\n",
+    "\n",
+    "## Importing Necessary Libraries\n",
+    "Let's start by importing the necessary libraries."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from ax.service.ax_client import AxClient, ObjectiveProperties\n",
+    "from ax.utils.notebook.plotting import render\n",
+    "from ax.service.utils.report_utils import exp_to_df\n",
+    "from submitit import AutoExecutor, LocalJob, DebugJob\n",
+    "import time"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Defining the Function to Optimize\n",
+    "We'll define a simple function to optimize. This function takes two parameters, and returns a single metric."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def evaluate(parameters):\n",
+    "    x = parameters[\"x\"]\n",
+    "    y = parameters[\"y\"]\n",
+    "    return {\"result\": (x - 3)**2 + (y - 4)**2}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Note 1: SubmitIt's [CommandFunction](https://github.com/facebookincubator/submitit/blob/main/docs/examples.md#working-with-commands) allows you to define commands to run on the node and then redirects the standard output.\n",
+    "\n",
+    "Note 2: If you are using Hydra to manage configs, SubmitIt also has [Hydra integration](https://hydra.cc/docs/plugins/submitit_launcher/).\n",
```

Comment:
This is true, but this tutorial doesn't cover overriding hydra params using Ax trial configurations. So maybe should either add an example of that (here or to another tutorial), or maybe should just remove this note?

Reply:
Removed note, you are right that it doesn't fit here.